core/src/accounts_hash_verifier.rs

```diff
@@ -96,49 +96,73 @@ impl AccountsHashVerifier {
         pending_snapshot_package: Option<&PendingSnapshotPackage>,
         hashes: &mut Vec<(Slot, Hash)>,
         exit: &Arc<AtomicBool>,
-        fault_injection_rate_slots: u64,
+        fault_injection_rate_slots: Slot,
         snapshot_config: Option<&SnapshotConfig>,
         thread_pool: Option<&ThreadPool>,
     ) {
-        let snapshot_package =
-            snapshot_utils::process_accounts_package(accounts_package, thread_pool, None);
-        Self::process_snapshot_package(
-            snapshot_package,
+        Self::verify_accounts_package_hash(&accounts_package, thread_pool);
+
+        Self::push_hashes_to_cluster(
+            &accounts_package,
             cluster_info,
             trusted_validators,
             halt_on_trusted_validator_accounts_hash_mismatch,
-            pending_snapshot_package,
             hashes,
             exit,
             fault_injection_rate_slots,
-            snapshot_config,
+        );
+
+        Self::handle_snapshot(accounts_package, pending_snapshot_package, snapshot_config);
+    }
+
+    fn verify_accounts_package_hash(
+        accounts_package: &AccountsPackage,
+        thread_pool: Option<&ThreadPool>,
+    ) {
```

Comment:
This function used to basically be the code inside `snapshot_utils::process_accounts_package()`, but it only really did anything for AccountsHashVerifier, since `expected_hash` was `None` for all other uses. So now it only lives here :)